ci/test-clang-docker.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ ! -d /repo ]]; then
+    echo "[-] This script needs to be run within Docker! Use test-clang.sh instead."
+    exit 1
+fi
+
+clang=/repo/toolchain/bin/clang
+hello_c=/repo/llvm-project/hello.c
+hello_cpp=/repo/llvm-project/hello.cpp
+hello_exe=/tmp/hello
+host_arch=$(uname -m)
+
+# Print clang information, which makes sure it can run
+echo "[+] Testing 'clang --version'"
+"$clang" --version
+
+case "$(source /usr/lib/os-release; echo "$ID")" in
+    fedora)
+        musl_cc_flags=(
+            -B /usr/"$host_arch"-linux-musl/lib64   # Scrt1.o, crti.o, crtn.o
+            -I /usr/"$host_arch"-linux-musl/include # stdio.h
+            -L /usr/"$host_arch"-linux-musl/lib64   # -lc
+        )
+
+        echo "[+] Updating OS"
+        dnf update -qy
+
+        echo "[+] Installing musl"
+        dnf install -qy \
+            musl-devel \
+            musl-libc \
+            musl-libc-static
+        ;;
+esac
+
+########
+# MUSL #
+########
+
+echo "[+] Testing dynamically linking hello.c against musl"
+"$clang" "${musl_cc_flags[@]}" -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
+echo "[+] Testing statically linking hello.c against musl"
+"$clang" "${musl_cc_flags[@]}" -static -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
+echo "[+] Testing dynamically linking hello.cpp against musl"
+# --rpath to avoid having to pull in libc++ and libunwind from distribution
+"$clang"++ "${musl_cc_flags[@]}" -Wl,--rpath=/repo/toolchain/lib/"$host_arch"-alpine-linux-musl -o "$hello_exe" "$hello_cpp"
```

Comment:
`++` on the outside is ghetto. How about just defining `$CC` and `$CXX`?

Reply:
> `++` on the outside is ghetto.

Hah, I did think it was a little hacky :)

> How about just defining `$CC` and `$CXX`?

Sure, I can do that!

Reply:
Done: 55125a5086cd353263a06b1917736fe2ead0d194